historical_system_profiles/archiver.py

```diff
@@ -81,6 +83,36 @@ def _archive_profile(data, ptc, logger):
             "wrote %s to historical database (inv id: %s, captured_on: %s)"
             % (hsp.id, hsp.inventory_id, hsp.captured_on)
         )
+    # After the new hsp is saved, we need to check for any reason to alert Notifications
+    _check_need_for_notifications(host["id"])
```

Comment:
I am not really sure we need to do that for both branches of if-else (lines 69 and 77) as with duplicate one we don't write anything into db only with new one we do.